shared/question/question.ts

```diff
@@ -48,7 +48,7 @@ class Question {
     }
 
     isAnswerValid() {
-        return !(this.required && this.answer === undefined);
+        return !(this.required && (this.answer === undefined || (this.answer.length !== undefined && !this.answer.length)));
```

Comment:
```suggestion
        return !(this.required && (this.answer === undefined || this.answer.length === 0));
```
Also, I think this check should be only on question types which their answer's type is an array (`CheckboxQuestion`)

Reply:
This is not right since it works also for open questions too, that is why I put it here :) 
in and case when the answer is a string + required, it is better to check that